dispatch/static/manager/src/js/components/fields/IntegerField.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+
+import { TextInput } from '../inputs'
+
+export default class IntegerField extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      selectionStart: 0,
+      selectionEnd: 0
+    }
+  }
+
+  onChange(val) {
+    let { selectionStart, selectionEnd } = this.refs.input.refs.input
+    const initialLength = val.length
+
+    val = val.replace(/[^\d-]+/g, '')
+
+    // allow a '-' at start, but nowhere else
+    if (val[0] == '-') {
+      val = `-${val.substr(1).replace(/-/g, '')}`
+    } else {
+      val = val.replace(/-/g, '')
+    }
+
+    // the change in position of the caret is equal to the difference
+    // in the length of the value
+    const dLength = val.length - initialLength
+    selectionStart += dLength
+    selectionEnd += dLength
+
+    this.setState({ selectionStart, selectionEnd }, () => this.props.onChange(val))
+  }
+
+  componentDidUpdate() {
+    //put the caret back in the correct place
+    this.refs.input.refs.input.setSelectionRange(
+      this.state.selectionStart,
+      this.state.selectionEnd)
+  }
+
+  render() {
+    return (
+      <TextInput
+        ref='input'
+        placeholder={this.props.label}
+        value={this.props.data || ''}
+        fill={true}
+        type="text"
```

Comment:
I tried step. It makes little arrows that appear at the right hand side of the input field, in addition to the arrow keys, (de)increment the value by step. However floating point values can still be entered manually.

Perhaps I am being overly pedantic; users will probably know they can't enter a fraction of an event and it will get caught by the server's field validation anyway.